### PR TITLE
Add Magnum  qa test scenario.

### PR DESCRIPTION
**Pull request by gosipyan**

### CODE REVIEW

File: scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    name: cloud-mkphyscloud-qa-scenario-6a
+    node: cloud-mkphyscloud-gate-qa
+    description: |
+      <b>This job is managed by JJB! Changes must be done in
+      <a href='https://github.com/SUSE-Cloud/automation/tree/master/scripts/jenkins/qa/'>git</a>
+      </b>
+
+      This job will redeploy scenario-6a:
+        - 7 nodes
+        - HA with ipm (3 nodes)
```
Comment (aplanas) [inline scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml]: impi

File: scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        - Sharerd storage on a NFS server (uses qa-provided nfs shares)
```
Comment (aplanas) [inline scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml]: Shared?

File: scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+          # wipe out shared NFS that should be used by this deployment
+          ssh root@localhost "wipe_nfs_shares qa$hw_number"
+
+          # rest of code runs on admin node:
+          env | grep -e networking -e libvirt -e cloud > mkcloud.config
+          scp mkcloud.config root@$admin:
+          ret=0
+
+          ssh root@$admin "
```
Comment (aplanas) [inline scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml]: I am a bit lost with this amount of commands that we send via ssh. Looks like that we need to scape some quotes, and not sure where it ends.

Can we use this << EOF stream? http://stackoverflow.com/questions/4412238/whats-the-cleanest-way-to-ssh-and-run-multiple-commands-in-bash

File: scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+          export clusterconfig=$clusterconfig ;
+          export want_magnum=1 ;
+          export want_ceph=0
+          export runner_url=$runner_url ;
+          export scenario_url=$scenario_url ;
+          export want_node_aliases=controller=3,kvm=2,docker=2
+          export want_node_roles=controller=3,compute=4
+          export ostestroptions='" --regex '^magnum.tests.functional.api'"'
+          export scenario=/root/scenario.yml;
+          export commands=\"$commands\" "'
```
Comment (aplanas) [inline scripts/jenkins/qa/cloud-mkphyscloud-qa-scenario-6a.yaml]: like this "' at the end

**Comment by gosipyan** [comment]
done !

**Comment by toabctl** [comment]
+1

**Comment by vmoravec** [comment]
I have tested the scenario with jjb and found some problems:
* missing `bootstrapcrowbar` step in the commands
* wrong indentation for the ssh commands using `<<EOF`
* old freshadminvm binary

All are addresed in my branch here: https://github.com/vmoravec/automation/commit/922fca675bf91f96f05cd8d979aec7e668c7f2f1  . 
The build for this scenario is running here: https://ci.suse.de/view/Cloud/view/QA/job/cloud-mkphyscloud-qa-scenario-6a/2/console

File: scripts/scenarios/cloud7/qa/no-ssl/qa-scenario-6a.yaml [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+- barclamp: trove
+  deployment:
+    elements:
+      trove-server:
+      - cluster:services
+- barclamp: magnum
+   attributes:
+   deployment:
+    elements:
```
Comment (vmoravec) [inline scripts/scenarios/cloud7/qa/no-ssl/qa-scenario-6a.yaml]: Wrong indentation, 2 spaces are required. And I think the `attributes` section is not needed two lines above.

File: scripts/scenarios/cloud7/qa/no-ssl/qa-scenario-6a.yaml [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+      - "@@controller1@@"
+      - "@@controller2@@"
+      - "@@controller3@@"
+
+- barclamp: nfs_client
+  name: glance
+  attributes:
+    exports:
+      images:
+        nfs_server: 10.162.26.129
```
Comment (vmoravec) [inline scripts/scenarios/cloud7/qa/no-ssl/qa-scenario-6a.yaml]: Wrong IP, the job is preparing nfs shares on 10.162.24.1, see here: https://github.com/SUSE-Cloud/automation/pull/1181/files#diff-56165da18ec5aff63391672593647341R139

**Comment by gosipyan** [comment]
@vmoravec  codecliemate fail but i don’t get reason.I think its because #TODO  comment in code but I am not sure.

**Comment by vmoravec** [comment]
+1

**Comment by aplanas** [comment]
+1

**Pull request merged**